client/landing/gutenboarding/onboarding-block/edit.tsx

```diff
@@ -14,13 +14,21 @@ import { STORE_KEY } from '../store';
 import './style.scss';
 
 export default function OnboardingEdit() {
+	const [ vertical, setVertical ] = useState( '' );
 	const { siteTitle, siteType } = useSelect( select => select( STORE_KEY ).getState() );
+	const verticals = useSelect( select => select( STORE_KEY ).getVertical( vertical ) );
```

Comment:
If you observe the API requests, we get a single vertical request for `""`, which is all the verticals 🙂 That works well with the `FormTokenField` suggestions because it handles the filtering.

The intention with the resolver was to get the results by search term, but it may work better to just fetch them all up front and filter the results like we're doing now. This is worth investigating.

Reply:
Simplify state:
https://github.com/Automattic/wp-calypso/pull/37313